smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -329,13 +329,7 @@ private boolean writeRequestQueryString(
                     Shape target = model.expectShape(binding.getMember().getTarget());
                     String queryValue = getInputValue(context, binding.getLocation(), "input." + memberName,
                             binding.getMember(), target);
-                    if (target instanceof CollectionShape) {
-                        writer.write("query[__extendedEncodeURIComponent($S)] = $L.map(entry => "
-                                + "__extendedEncodeURIComponent(entry));", binding.getLocationName(), queryValue);
-                    } else {
-                        writer.write("query[__extendedEncodeURIComponent($S)] = __extendedEncodeURIComponent($L);",
-                                binding.getLocationName(), queryValue);
-                    }
+                    writer.write("query['$L'] = $L;", binding.getLocationName(), queryValue);
```

Comment:
Can this still use `$S` instead of `'$L'` ?

Reply:
Fixed in c619ba3